ENIGMAsystem/SHELL/Widget_Systems/Win32/dialogs.cpp

```diff
@@ -52,24 +62,39 @@ static string gs_but1, gs_but2, gs_but3;
 #include "Universal_System/debugscope.h"
 #endif
 
-void show_error(string errortext, const bool fatal)
-{
+void show_error(string errortext, const bool fatal) {
   #ifdef DEBUG_MODE
     errortext += enigma::debug_scope::GetErrors();
   #else
-  errortext = "Error in some event or another for some object: \r\n" + errortext;
+    errortext = "Error in some event or another for some object: \r\n" + errortext;
   #endif
 
-  if (MessageBox(NULL,errortext.c_str(),"Error",MB_ABORTRETRYIGNORE | MB_ICONERROR)==IDABORT)
-    exit(0);
+  string strStr = errortext;
+  string strWindowCaption = "Error";
 
-  if (fatal)
-    printf("FATAL ERROR: %s\n",errortext.c_str()),
-    exit(0);
+  if (strStr != "")
+    strStr = strStr + "\n\n";
+
+  if (!fatal)
+    strStr = strStr + "Do you want to abort the application?";
   else
-    printf("ERROR: %s\n",errortext.c_str());
+    strStr = strStr + "Click 'OK' to abort the application.";
 
-  //ABORT_ON_ALL_ERRORS();
+  tstring tstrStr = widen(strStr);
+  tstring tstrWindowCaption = widen(strWindowCaption);
+
+  double result;
+
+  if (!fatal) {
+    result = MessageBoxW(enigma::hWnd, tstrStr.c_str(), tstrWindowCaption.c_str(), MB_YESNO | MB_ICONERROR | MB_DEFBUTTON1 | MB_APPLMODAL);
+    printf("ERROR: %s\n", errortext.c_str());
+  } else {
+    result = MessageBoxW(enigma::hWnd, tstrStr.c_str(), tstrWindowCaption.c_str(), MB_OK | MB_ICONERROR | MB_DEFBUTTON1 | MB_APPLMODAL);
+    printf("FATAL ERROR: %s\n", errortext.c_str());
+  }
```

Comment:
Please start making regular use of the ternary operator `a ? b : c` from here on. Use my shorter version of this if-else block please:
```cpp
DWORD flags = MB_ICONERROR | MB_DEFBUTTON1 | MB_APPLMODAL;
result = MessageBoxW(enigma::hWnd, tstrStr.c_str(), tstrWindowCaption.c_str(), flags | (fatal ? MB_OK : MB_YESNO));
printf(fatal ? "FATAL ERROR: %s\n" : "ERROR: %s\n", errortext.c_str());
```

Reply:
Will do!

Reply:
@RobertBColton Are you saying you want me to change something here Again, or are you trying to say something else?

Reply:
No, I was just reiterating so you understood you shouldn't go wild & crazy with ternarys just because I said to use them. In the same code block you should limit yourself to checking the same boolean at most two times with a ternary, otherwise you should extract helper methods. Just a rule of thumb.